containerz/containerz.proto

```diff
@@ -64,6 +80,24 @@ service Containerz {
   // (i.e. kill) a container.
   rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
 
+  // UpdateContainer updates a running container to the image specified in the
+  // request. By default the operation is synchronous which means that the
+  // request will only return once the container has either been successfully
+  // updated or the update has failed. If the client requests an asynchronous
+  // update then the server must perform all validations (e.g. does the
+  // requested image exist on the system or does the instance name exist) and
+  // return to the client and the update happens asynchronously. If the update
+  // fails, the server must restore the previous version of the container. It
+  // is up to the client to check if the update actually updates the container
+  // to the requested version or not.
+  // If no StartContainerRequest is provided, the server must reuse the same
```

Comment:
Is there a reason not to just require the `StartContainerRequest`? Can we also clarify that this is the `params` field in the `UpdateContainerRequest` in this RPC?

Reply:
In some cases we want to start the container with exactly the same resources as the previous container. It seems easier to do this by having the server inspect the running container, copy values and update the container. That being said - I am happy to make this field required.

Reply:
Per discussion -- I'd prefer this is declarative so that we don't need to find the place where the container was created.